text/0017-shared-config.md

```diff
@@ -0,0 +1,120 @@
+# Configuration
+- Start Date: 2019/11/11
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/16
+
+## Definitions
+
+**Config files** allow implementers to customize the behavior of microfrontends
+by providing static files. Common formats include JSON, YAML, and HCL.
+
+A **config schema** is the set of expectations that a microfrontend has of
+implementation-provided config files. It says how the config file must be 
+structured, what keys it must have, and constrains what types of values are allowed.
+
+A **config library** is a code module that allows developers to define config 
+schemas, and loads config files in accordance with that schema. It may also 
+provide other tooling related to configuration.
+
+**Shared config** is configuration values that are shared between distinct modules.
+
+
+## Decision
+
+Config elements that will be shared between multiple modules should be added to a `common`
```

Comment:
I'm nervous about this. Common means that there's no "owner" of it - no code and no person who actually understands all of it since it's shared across all codebases and organizations. Seems like an organizational bottleneck to me to try to maintain it.

Reply:
This is a very legitimate concern.

Reply:
Added a line below specifying that this would live in `esm-module-config` and be maintained by its maintainer. That wouldn't make it less of a bottleneck though. But I think the bottleneck is inevitable for something intended to be a shared resource like this.

Reply:
👍